### PR TITLE
kommander: Fix missing kubecost cm on upgrade

### PR DESCRIPTION
**Pull request by gracedo**
**What type of PR is this?**
Bug

**What this PR does/ why we need it**:
We saw upgrade tests beginning to fail in the konvoy nightlies after removing the `pre-upgrade` hook to the job that creates the `kubecost-thanos-query-stores` cm. I've fixed this to instead run a bash script that checks whether the cm exists or not and only creates it if it does. I applied the same changes to the centralized thanos query stores cm job as well since the same problem applies.

**Which issue(s) this PR fixes**:
relates to D2IQ-69505

**Special notes for your reviewer**:
I tested this using my kubeaddons-kommander branch `test-kubecost` to deploy this updated version of the chart. I tested a fresh install (job pod logs):
```
checking if kubecost-thanos-query-stores configmap exists
Error from server (NotFound): configmaps "kubecost-thanos-query-stores" not found
kubecost-thanos-query-stores configmap does not exist - creating
configmap/kubecost-thanos-query-stores created
```
as well as an upgrade where the cm already exists:
```
checking if kubecost-thanos-query-stores configmap exists
kubecost-thanos-query-stores configmap already exists - no need to create
```

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: stable/kommander/templates/kubecost/kubecost-query-stores-configmap.yaml [hunk 1]
```diff
@@ -4,19 +4,48 @@
...
+          - "-c"
+          - |
+            /bin/bash <<'EOF'
+            set -o nounset
+            set -o errexit
+            set -o pipefail
+
+            echo "checking if {{ . }} configmap exists"
+
+            RES=$(kubectl get configmap {{ . }} || true)
```
Comment (dlipovetsky) [inline stable/kommander/templates/kubecost/kubecost-query-stores-configmap.yaml:39]: Do you want the script to fail if `kubectl get configmap` fails? If you do, I believe you will need to enable `pipefail` within the subshell. If you just want to ignore a 404 (which is normal, if the configmap wasn't created in the previous version of the Chart, and is to be created on upgrade), then I recommend using `--ignore-not-found=true`.

```suggestion
            RES=$(set -o errexit; set -o pipefail; kubectl get --ignore-not-found=true configmap {{ . }})
```

File: stable/kommander/charts/kommander-thanos/templates/query-stores-configmap.yaml [hunk 2]
```diff
@@ -4,19 +4,48 @@
...
+          - "-c"
+          - |
+            /bin/bash <<'EOF'
+            set -o nounset
+            set -o errexit
+            set -o pipefail
+
+            echo "checking if {{ . }} configmap exists"
+
+            RES=$(kubectl get configmap {{ . }} || true)
```
Comment (dlipovetsky) [inline stable/kommander/charts/kommander-thanos/templates/query-stores-configmap.yaml:39]: Same as my other comment

File: stable/kommander/templates/kubecost/kubecost-query-stores-configmap.yaml (hunk 1 above)
Comment (gracedo) [reply stable/kommander/templates/kubecost/kubecost-query-stores-configmap.yaml:39]: good call! I'll pass in the `ignore-not-found` flag which is cleaner :) (`set -o errexit` and `set -o pipefail` are set at the top of script)

Comment (dlipovetsky) [reply stable/kommander/templates/kubecost/kubecost-query-stores-configmap.yaml]: I think we were both a little wrong about the bash options :joy:

Bash subshells do not inherit options intuitively.

For years, I had thought that a command-substitution subshell does not inherit `pipefail`, but in fact it does, so it is not necessary to set it in any subshell.

On the other hand, a command-substitution subshell does not inherit `errexit`, so it is necessary to set it in every subshell. (There are two workarounds: Use bash in its non-default POSIX mode, or use bash >= 4.4 and set the `inherit_errexit` option in the containing shell.)

For additional details and some good examples, please see https://dougrichardson.us/2018/08/03/fail-fast-bash-scripting.html

Comment (dlipovetsky) [reply stable/kommander/templates/kubecost/kubecost-query-stores-configmap.yaml]: I think that, if we don't set `errexit` in the subshell, then, if kubectl throws an error for some reason (e.g. network issue), then the script will drop down to "configmap already exists."

**Pull request merged**